src/main/java/com/unstoppabledomains/config/client/Client.java

```diff
@@ -19,9 +19,23 @@ public static String getVersion() {
 
     private static String initClientVersion() {
         final InputStreamReader reader = new InputStreamReader(Client.class.getResourceAsStream(CLIENT_FILE));
-        final String jsonString = new BufferedReader(reader).lines().collect(Collectors.joining("\n"));
-        final JsonObject jsonObj = new Gson().fromJson(jsonString, JsonObject.class);
-
-        return jsonObj.get("version").getAsString();
+        final BufferedReader buffer = new BufferedReader(reader);
+        try {
+            String jsonString = "";
+            String line;
+            // We use readLine instead of lines 
+            // because some of the clients had issues with old Android devices
+            line = buffer.readLine();
+            while ( line != null) {
+                jsonString = jsonString.concat(line);
+                line = buffer.readLine();
+            }
+            final JsonObject jsonObj = new Gson().fromJson(jsonString, JsonObject.class);
+            return jsonObj.get("version").getAsString();
+        } catch (IOException e) {
+            e.printStackTrace();
+            return "-2";
```

Comment:
What does -2 means? It's better to name a variable or constant which will describe -2 meaning.

Reply:
It has no meaning at all. This event will never happen as we bundle the config .json file with the library. If for some reason the config file is not accessible (corrupted .jar file) it will break later on as there is no information to what contract should we make the call for resolution

Reply:
I think we can try it like this: 
```suggestion
            private static String initClientVersion() {
        final JsonReader jsonReader =
                new JsonReader(new InputStreamReader(Client.class.getResourceAsStream(CLIENT_FILE)));
        final JsonObject jsonObj = new Gson().fromJson(jsonReader, JsonObject.class);

        return jsonObj.get("version").getAsString();
    }
```